src/main/java/seedu/partyplanet/logic/commands/DeleteContactWithTagCommand.java

```diff
@@ -14,88 +13,85 @@
 
 /**
  * Deletes all persons, that is tagged with the target tags, from PartyPlanet.
- * Provided that they do not have other tags.
  */
 public class DeleteContactWithTagCommand extends DeleteCommand {
 
-    public static final String MESSAGE_DELETE_TAGS_SUCCESS = "Deleted tags : %1$s";
+    public static final String MESSAGE_PERSON_NOT_REMOVED_ANY =
+            "These tags do not exist in persons listed. No person removed.";
+    public static final String MESSAGE_PERSON_NOT_REMOVED_ALL =
+            "These combination of tags do not exist in persons listed. No person removed.";
 
     private final Set<Tag> targetTags;
 
     private final List<Person> deletedPersons;
+    private final boolean isAny;
 
     /**
      * Creates an DeleteContactWithTagCommand to delete the {@code Person} with specified {@code Tag}
      */
-    public DeleteContactWithTagCommand(Set<Tag> targetTags) {
+    public DeleteContactWithTagCommand(Set<Tag> targetTags, boolean isAny) {
+        assert targetTags.size() > 0;
         this.targetTags = targetTags;
         deletedPersons = new ArrayList<>();
+        this.isAny = isAny;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        List<Person> personList = model.getPersonListCopy();
+        List<Person> personList = model.getFilteredPersonList();
 
-        for (Person person : personList) {
-            removePersonWithTags(model, person);
+        if (isAny) {
+            for (Person person : personList) {
+                if (containsAnyTags(person)) {
+                    deletedPersons.add(person);
+                }
+            }
```

Comment:
There is an existing predicate for tag testing, if intending to integrate with the existing `Person` model. Will require addition of a new predicate to separately check for all tags.

```suggestion
    for (Tag t: targetTags) {
        Predicate<Person> predicate = new TagsContainExactTagPredicate(t.tagName);
        for (Person person: personList) {  // this can be refactored into separate method as usual
            if (predicate.test(person)) {
                deletedPersons.add(person);
            }
        }
    }
```

Reply:
Could not find a simple way to make tags the outer loop for `containAllTags()`.
So kept persons as the outer loop for both All and Any, for readability.
However, made use of `TagsContainsExactTagPredicate` as suggested for better abstraction (demeter law)